mobile/library/swift/stats/Element.swift

```diff
@@ -9,13 +9,27 @@ public final class Element: NSObject, ExpressibleByStringLiteral {
   internal let value: String
 
   public init(stringLiteral value: String) {
-    guard value.range(of: kPattern, options: .regularExpression) != nil else {
-      preconditionFailure("Element values must conform to the regex '\(kPattern)'.")
-    }
+    precondition(
+      value.matchesStatsElementPattern,
+      "Element values must conform to the regex '\(kPattern)'."
+    )
     self.value = value
   }
 
   public override func isEqual(_ object: Any?) -> Bool {
     return (object as? Element)?.value == self.value
   }
 }
+
+extension String {
+  var matchesStatsElementPattern: Bool {
+    if #available(iOS 16.0, macOS 13.0, *) {
+      return self.contains(/^[A-Za-z_]+$/)
+    }
+#if canImport(std)
+    return (self as NSString).range(of: kPattern, options: regularExpression).location != NSNotFound
+#else
+    return self.range(of: kPattern, options: .regularExpression) != nil
```

Comment:
not sure whether kPattern should be declared at the top of the file if it's being effectively overriden by `/^[A-Za-z_]+$/` in some cases in this method anyway. Maybe it would make it easier to parse the code if all constants/regexes were defined somewhere in the body of `matchesStatsElementPattern` property

Reply:
I can inline it in the `matchesStatsElementPattern` body, but then we can't include it in the precondition failure message on line 14. 😞 

Alternatively instead of a regex literal, we could parse it at runtime with `self.contains(try Regex(kPattern))`, but that has higher overhead than compiling the regex into optimized machine code at compile time.

Alternatively pt2 is to remove the entire `self.contains(/^[A-Za-z_]+$/)` codepath and add a todo comment to use regex literals when we can target iOS 16 as our min iOS version.

---

Overall my preference is to keep it as-is, even if it means some duplication. It's not like we update this code very often.